### PR TITLE
use pdo for postgres setup

### LINKED ISSUE #367: PostgreSQL: installation wizzard claims both the vendor specific and PDO driver are needed
Using PostgreSQL as backend the installation wizzard claims that both the PDO driver ("pdo_pgsql") and the vendor specific driver ("pgsql") are needed. To my knowledge the two drivers are never used in parallel by a single PHP application so there seems to be something wrong either with the wizzard or the PostgreSQL backend itself.

If only `pdo_pgsql` is available the wizzard's initial page doesn't come up but an error message
nextcloud_postgresql-both-drivers_pdo_pgsql-only
If only `pgsql` is available the initial page does come up but after filing all credentials and clicking "Finish setup" there's
nextcloud_postgresql-both-drivers_pgsql-only
None of the two messages can be seen when both drivers are available.

Seen running 9.0.52 or 103417f on Arch Linux (PostgreSQL 9.5.3, PHP 7.0.8, Apache 2.4.23 using the PHP module).

### PR DESCRIPTION
**Pull request by icewind1991**
Removes the need of having the pgsql extension installed by using pdo/doctrine during setup

Fixes #367

### CODE REVIEW

**Comment by MorrisJobke** [comment]
```
There were 2 failures:

1) Test\SetupTest::testGetSupportedDatabasesWithNoWorking
OC\Setup::getAvailableDbDriversForPdo() was not expected to be called more than once.

/drone/src/github.com/nextcloud/server/lib/private/Setup.php:170
/drone/src/github.com/nextcloud/server/tests/lib/SetupTest.php:90

2) Test\SetupTest::testGetSupportedDatabasesWithAllWorking
OC\Setup::getAvailableDbDriversForPdo() was not expected to be called more than once.

/drone/src/github.com/nextcloud/server/lib/private/Setup.php:170
/drone/src/github.com/nextcloud/server/tests/lib/SetupTest.php:114
```

File: lib/private/AllConfig.php [hunk 1]
```diff
@@ -428,4 +428,8 @@ public function getUsersForUserValue($appName, $key, $value) {
 
 		return $userIDs;
 	}
+
+	public function getSystemConfig() {
```
Comment (nickvergessen) [inline lib/private/AllConfig.php]: :cold_sweat: why? just use the methods from above?

Comment (icewind1991) [inline lib/private/AllConfig.php]: The connectionfactory wants a systemconfig instance

Comment (nickvergessen) [inline lib/private/AllConfig.php]: Then mark `@internal` at least please

Comment (icewind1991) [inline lib/private/AllConfig.php]: It's not exposed in the public interface

**Comment by icewind1991** [comment]
Tests are all green now

**Comment by LukasReschke** [comment]
👍

File: lib/private/AllConfig.php (hunk 1 above)
Comment (rullzer) [inline lib/private/AllConfig.php]: mmmm Maybe we should change the connectionfactory at some point then.

**Comment by rullzer** [comment]
:+1:

File: lib/private/Setup/PostgreSQL.php [hunk 2]
```diff
@@ -72,106 +64,93 @@ public function setupDatabase($username) {
...
+	}
+
+	private function createDBUser(IDBConnection $connection) {
+		try {
+			if ($this->userExists($connection, $this->dbUser)) {
+				// change the password
+				$query = $connection->prepare("ALTER ROLE " . addslashes($this->dbUser) . " CREATEDB WITH PASSWORD " . addslashes($this->dbPassword));
+			} else {
+				// create the user
+				$query = $connection->prepare("CREATE USER " . addslashes($this->dbUser) . " CREATEDB PASSWORD " . addslashes($this->dbPassword));
```
Comment (nickvergessen) [inline lib/private/Setup/PostgreSQL.php]: This broke the tests on my app:
```
Syntax error: 7 ERROR:  syntax error at or near "zxngvsgscub2i9frs64ac9r43ivk6j"

LINE 1: CREATE USER oc_admin CREATEDB PASSWORD zxngvsgscub2i9frs64ac
```

The `'` are missing, I will add them.

https://travis-ci.org/nextcloud/announcementcenter/jobs/145487372

**Pull request merged**